<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-08-06" />

<title>ID-seq from FASTQ to count tables</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapter IDseq tech</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="y_chapter_main.html">Figs main</a>
</li>
<li>
  <a href="z_chapter_suppl.html">Figs suppl</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="e_IDseq_ann.html">Annotation</a>
    </li>
    <li>
      <a href="DS078_IDseq_analysis.html">DS078 analysis</a>
    </li>
    <li>
      <a href="DS091_IDseq_analysis.html">DS091 analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ID-seq from FASTQ to count tables</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-08-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapter_IDseq_tech/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240806code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240806)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240806code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240806)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong31f8444">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 31f8444
</a>
</p>
</div>
<div id="strongRepositoryversionstrong31f8444"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 31f8444.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/DS075_combined_analysis.R
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/BB014_LibraryIndex/
    Untracked:  data/DS027_FirstIDseq/
    Untracked:  data/DS055_LibraryKlenowAmp/
    Untracked:  data/DS056_LibraryAmp1/
    Untracked:  data/DS059_LibraryByproducts/
    Untracked:  data/DS072_LibraryAmp2/
    Untracked:  data/DS075_BarcodeCheck/
    Untracked:  data/DS078_Blocking1/
    Untracked:  data/DS080_Blocking2/
    Untracked:  data/DS084_BioInhibition/
    Untracked:  data/DS091_ImmunostainingBio/
    Untracked:  data/DSxxx_Conjugations/
    Untracked:  data/annotated_gels_batch1-10_digital.pptx
    Untracked:  output/BioAnalyzer_plots/
    Untracked:  output/DS075_BarcodeCheck/
    Untracked:  output/DS078_Blocking1/
    Untracked:  output/DS091_ImmunostainingBio/
    Untracked:  output/figures/
    Untracked:  output/network_visual/

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/c_flow_vis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/d_IDseq_count.Rmd</code>) and
HTML (<code>docs/d_IDseq_count.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
31f8444
</td>
<td>
mwitmond
</td>
<td>
2025-02-10
</td>
<td>
Revised ch3 figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
a92dfa0
</td>
<td>
mwitmond
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
4e7f883
</td>
<td>
mwitmond
</td>
<td>
2024-08-06
</td>
<td>
Publish initial project files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<p>Note: Run all bash code in the terminal (Ubuntu on Windows)</p>
<pre class="r"><code>library(rjson)
library(kableExtra)
library(dplyr)</code></pre>
<p>Install the conda environment from a .yml file</p>
<pre class="bash"><code># Check the directories for conda environments
conda config --show envs_dirs

# Move to directory with conda_environment.yml file:
cd /mnt/c/Users/meldew/&#39;OneDrive - Radboud Universiteit&#39;/Documenten/data/analysis/template_data_analysis_MW/data/config

# Install conda environment (done only once)
conda env create -f conda_environment.yml
# Environment name: bulkIDseq_kite
# Location: /home/meldewitmond/.conda/envs/bulkIDseq_kite</code></pre>
</div>
<div id="ds075-barcodecheck" class="section level2">
<h2>DS075: BarcodeCheck</h2>
<div id="fastq-to-counts" class="section level3">
<h3>FASTQ to counts</h3>
<p>Activate the conda environment from a .yml file</p>
<pre class="bash"><code># Activate the correct conda environment
conda activate bulkIDseq_kite

# Move to directory with all configuration files:
cd /mnt/c/Users/meldew/&#39;OneDrive - Radboud Universiteit&#39;/Documenten/data/analysis/chapter_IDseq_tech/data/DS075_BarcodeCheck/config</code></pre>
<p>Step 1: Prepare index and mismatch files to use <a
href="https://github.com/pachterlab/kite">kite</a> from pachterlab.</p>
<p>Use the manually created FeatureBarcodes.csv file,
custom_whitelist.txt file and other config files:
Huck/Cells/DynSign/GeneralProtocols/general_sequencing_metadata_files/config</p>
<div id="platea" class="section level4">
<h4>plateA</h4>
<pre class="bash"><code># Prepare folders for processing in the data folder:
cd ..
mkdir counts
cd counts
mkdir plateA
cd plateA</code></pre>
<pre class="bash"><code># Create the pseudoalignment of the reads (including time of running in the output)
# Adjust the .fastq.gz file names
time kallisto bus -i ../../config/FeaturesMismatch.idx -o ./ -x 10xv3 -t 4 ../../FASTQ/MW_DS075_lib1_plateA_A39_S27_R1_001.fastq.gz ../../FASTQ/MW_DS075_lib1_plateA_A39_S27_R2_001.fastq.gz

# Make the corrected bus
time bustools correct -w ../../config/custom_whitelist.txt output.bus -o output_corrected.bus

# Sort the corrected bus (can take very long with RNA seq; more reads = more time)
time bustools sort -t 4 -o output_sorted.bus output_corrected.bus

# Make count table (.mtx file)
mkdir featurecounts
time bustools count -o featurecounts/featurecounts --genecounts -g ../../config/FeaturesMismatch.t2g -e ./matrix.ec -t ./transcripts.txt ./output_sorted.bus
# Location of count tables: data\counts\plateA\featurecounts

cd ..</code></pre>
</div>
<div id="plateb" class="section level4">
<h4>plateB</h4>
<pre class="bash"><code># Prepare folders for processing in the data folder:
# cd ..
# mkdir counts
# cd counts
mkdir plateB
cd plateB</code></pre>
<pre class="bash"><code># Create the pseudoalignment of the reads (including time of running in the output)
# Adjust the .fastq.gz file names
time kallisto bus -i ../../config/FeaturesMismatch.idx -o ./ -x 10xv3 -t 4 ../../FASTQ/MW_DS075_lib2_plateB_A40_S28_R1_001.fastq.gz ../../FASTQ/MW_DS075_lib2_plateB_A40_S28_R2_001.fastq.gz

# Make the corrected bus
time bustools correct -w ../../config/custom_whitelist.txt output.bus -o output_corrected.bus

# Sort the corrected bus (can take very long with RNA seq; more reads = more time)
time bustools sort -t 4 -o output_sorted.bus output_corrected.bus

# Make count table (.mtx file)
mkdir featurecounts
time bustools count -o featurecounts/featurecounts --genecounts -g ../../config/FeaturesMismatch.t2g -e ./matrix.ec -t ./transcripts.txt ./output_sorted.bus
# Location of count tables: data\counts\plateA\featurecounts

cd ..</code></pre>
</div>
</div>
<div id="run-log" class="section level3 tabset">
<h3 class="tabset">Run log</h3>
<div id="platea-1" class="section level4">
<h4>plateA</h4>
<pre class="r"><code># Print info on count generation run
PROT_log&lt;- rjson::fromJSON(file = &quot;data/counts/plateA/run_info.json&quot;)
PROT_log&lt;- do.call(rbind, lapply(PROT_log, data.frame))
colnames (PROT_log) &lt;- &quot;&quot;
PROT_log %&gt;%
  kbl() %&gt;%
  kable_styling()</code></pre>
</div>
<div id="plateb-1" class="section level4">
<h4>plateB</h4>
<pre class="r"><code># Print info on count generation run
PROT_log&lt;- rjson::fromJSON(file = &quot;data/counts/plateB/run_info.json&quot;)
PROT_log&lt;- do.call(rbind, lapply(PROT_log, data.frame))
colnames (PROT_log) &lt;- &quot;&quot;
PROT_log %&gt;%
  kbl() %&gt;%
  kable_styling()</code></pre>
</div>
</div>
<div id="output" class="section level3">
<h3>Output</h3>
<p>Unfiltered counts are stored (for each sample) in
data/counts/plate/featurecounts/ folder as .mtx format (sparse
matrix).</p>
</div>
</div>
<div id="ds078-blocking1" class="section level2">
<h2>DS078: Blocking1</h2>
<div id="fastq-to-counts-1" class="section level3">
<h3>FASTQ to counts</h3>
<p>Activate the conda environment from a .yml file</p>
<pre class="bash"><code># Activate the correct conda environment
conda activate bulkIDseq_kite

# Move to directory with all configuration files:
cd /mnt/c/Users/meldew/&#39;OneDrive - Radboud Universiteit&#39;/Documenten/data/analysis/chapter_IDseq_tech/data/DS078_Blocking1/config</code></pre>
<p>Step 1: Prepare index and mismatch files to use <a
href="https://github.com/pachterlab/kite">kite</a> from pachterlab.</p>
<p>Use the manually created FeatureBarcodes.csv file,
custom_whitelist.txt file and other config files:
Huck/Cells/DynSign/GeneralProtocols/general_sequencing_metadata_files/config</p>
<div id="platea-2" class="section level4">
<h4>plateA</h4>
<pre class="bash"><code># Prepare folders for processing in the data folder:
cd ..
mkdir counts
cd counts
mkdir plateA
cd plateA</code></pre>
<pre class="bash"><code># Create the pseudoalignment of the reads (including time of running in the output)
# Adjust the .fastq.gz file names
time kallisto bus -i ../../config/FeaturesMismatch.idx -o ./ -x 10xv3 -t 4 ../../FASTQ/JvB_DynSign078_A40_S24_R1_001.fastq.gz ../../FASTQ/JvB_DynSign078_A40_S24_R2_001.fastq.gz

# Make the corrected bus
time bustools correct -w ../../config/custom_whitelist.txt output.bus -o output_corrected.bus

# Sort the corrected bus (can take very long with RNA seq; more reads = more time)
time bustools sort -t 4 -o output_sorted.bus output_corrected.bus

# Make count table (.mtx file)
mkdir featurecounts
time bustools count -o featurecounts/featurecounts --genecounts -g ../../config/FeaturesMismatch.t2g -e ./matrix.ec -t ./transcripts.txt ./output_sorted.bus
# Location of count tables: data\counts\plateA\featurecounts

cd ..</code></pre>
</div>
</div>
<div id="run-log-1" class="section level3 tabset">
<h3 class="tabset">Run log</h3>
<div id="platea-3" class="section level4">
<h4>plateA</h4>
<pre class="r"><code># Print info on count generation run
PROT_log&lt;- rjson::fromJSON(file = &quot;data/counts/plateA/run_info.json&quot;)
PROT_log&lt;- do.call(rbind, lapply(PROT_log, data.frame))
colnames (PROT_log) &lt;- &quot;&quot;
PROT_log %&gt;%
  kbl() %&gt;%
  kable_styling()</code></pre>
</div>
</div>
<div id="output-1" class="section level3">
<h3>Output</h3>
<p>Unfiltered counts are stored (for each sample) in
data/counts/plate/featurecounts/ folder as .mtx format (sparse
matrix).</p>
</div>
</div>
<div id="ds091-immunostainingbio" class="section level2">
<h2>DS091: ImmunostainingBio</h2>
<div id="fastq-to-counts-2" class="section level3">
<h3>FASTQ to counts</h3>
<p>Activate the conda environment from a .yml file</p>
<pre class="bash"><code># Activate the correct conda environment
conda activate bulkIDseq_kite

# Move to directory with all configuration files:
cd /mnt/c/Users/meldew/&#39;OneDrive - Radboud Universiteit&#39;/Documenten/data/analysis/chapter_IDseq_tech/data/DS091_ImmunostainingBio/config</code></pre>
<p>Step 1: Prepare index and mismatch files to use <a
href="https://github.com/pachterlab/kite">kite</a> from pachterlab.</p>
<p>Use the manually created FeatureBarcodes.csv file,
custom_whitelist.txt file and other config files:
Huck/Cells/DynSign/GeneralProtocols/general_sequencing_metadata_files/config</p>
<div id="platea-4" class="section level4">
<h4>plateA</h4>
<pre class="bash"><code># Prepare folders for processing in the data folder:
cd ..
mkdir counts
cd counts
mkdir plateA
cd plateA</code></pre>
<pre class="bash"><code># Create the pseudoalignment of the reads (including time of running in the output)
# Adjust the .fastq.gz file names
time kallisto bus -i ../../config/FeaturesMismatch.idx -o ./ -x 10xv3 -t 4 ../../FASTQ/42525_MW_DynSign091_A39_R1.fastq.gz ../../FASTQ/42525_MW_DynSign091_A39_R2.fastq.gz

# Make the corrected bus
time bustools correct -w ../../config/custom_whitelist.txt output.bus -o output_corrected.bus

# Sort the corrected bus (can take very long with RNA seq; more reads = more time)
time bustools sort -t 4 -o output_sorted.bus output_corrected.bus

# Make count table (.mtx file)
mkdir featurecounts
time bustools count -o featurecounts/featurecounts --genecounts -g ../../config/FeaturesMismatch.t2g -e ./matrix.ec -t ./transcripts.txt ./output_sorted.bus
# Location of count tables: data\counts\plateA\featurecounts

cd ..</code></pre>
</div>
</div>
<div id="run-log-2" class="section level3 tabset">
<h3 class="tabset">Run log</h3>
<div id="platea-5" class="section level4">
<h4>plateA</h4>
<pre class="r"><code># Print info on count generation run
PROT_log&lt;- rjson::fromJSON(file = &quot;data/counts/plateA/run_info.json&quot;)
PROT_log&lt;- do.call(rbind, lapply(PROT_log, data.frame))
colnames (PROT_log) &lt;- &quot;&quot;
PROT_log %&gt;%
  kbl() %&gt;%
  kable_styling()</code></pre>
</div>
</div>
<div id="output-2" class="section level3">
<h3>Output</h3>
<p>Unfiltered counts are stored (for each sample) in
data/counts/plate/featurecounts/ folder as .mtx format (sparse
matrix).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
