<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jessie van Buggenum" />

<meta name="date" content="2022-06-28" />

<title>DS078 IDseq analysis</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapter IDseq tech</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="y_chapter_main.html">Figs main</a>
</li>
<li>
  <a href="z_chapter_suppl.html">Figs suppl</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="e_IDseq_ann.html">Annotation</a>
    </li>
    <li>
      <a href="DS078_IDseq_analysis.html">DS078 analysis</a>
    </li>
    <li>
      <a href="DS091_IDseq_analysis.html">DS091 analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DS078 IDseq analysis</h1>
<h4 class="author">Jessie van Buggenum</h4>
<h4 class="date">2022-06-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapter_IDseq_tech/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240806code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240806)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240806code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240806)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong31f8444">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 31f8444
</a>
</p>
</div>
<div id="strongRepositoryversionstrong31f8444"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 31f8444.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/DS075_combined_analysis.R
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/BB014_LibraryIndex/
    Untracked:  data/DS027_FirstIDseq/
    Untracked:  data/DS055_LibraryKlenowAmp/
    Untracked:  data/DS056_LibraryAmp1/
    Untracked:  data/DS059_LibraryByproducts/
    Untracked:  data/DS072_LibraryAmp2/
    Untracked:  data/DS075_BarcodeCheck/
    Untracked:  data/DS078_Blocking1/
    Untracked:  data/DS080_Blocking2/
    Untracked:  data/DS084_BioInhibition/
    Untracked:  data/DS091_ImmunostainingBio/
    Untracked:  data/DSxxx_Conjugations/
    Untracked:  data/annotated_gels_batch1-10_digital.pptx
    Untracked:  output/BioAnalyzer_plots/
    Untracked:  output/DS075_BarcodeCheck/
    Untracked:  output/DS078_Blocking1/
    Untracked:  output/DS091_ImmunostainingBio/
    Untracked:  output/figures/
    Untracked:  output/network_visual/

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/c_flow_vis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DS078_IDseq_analysis.Rmd</code>)
and HTML (<code>docs/DS078_IDseq_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
31f8444
</td>
<td>
mwitmond
</td>
<td>
2025-02-10
</td>
<td>
Revised ch3 figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(DropletUtils)
library(Seurat)
library(Matrix)
library(tidyverse)
library(scico)
library(dplyr)
library(ggthemes)
library(kableExtra)
library(ggpubr)
library(patchwork)
library(readxl)
library(ggpubr)
   library(scales)
library(platetools)
library(viridis)
## From online; Slightly modified from BUSpaRse, just to avoid installing a few dependencies not used here
read_count_output &lt;- function(dir, name) {
  dir &lt;- normalizePath(dir, mustWork = TRUE)
  m &lt;- readMM(paste0(dir, &quot;/&quot;, name, &quot;.mtx&quot;))
  m &lt;- Matrix::t(m)
  m &lt;- as(m, &quot;dgCMatrix&quot;)
  # The matrix read has cells in rows
  ge &lt;- &quot;.genes.txt&quot;
  genes &lt;- readLines(file(paste0(dir, &quot;/&quot;, name, ge)))
  barcodes &lt;- readLines(file(paste0(dir, &quot;/&quot;, name, &quot;.barcodes.txt&quot;)))
  colnames(m) &lt;- barcodes
  rownames(m) &lt;- genes
  return(m)
}

## For re-ordered antibody plot
scale_x_reordered &lt;- function(..., sep = &quot;___&quot;) {
  reg &lt;- paste0(sep, &quot;.+$&quot;)
  ggplot2::scale_x_discrete(labels = function(x) gsub(reg, &quot;&quot;, x), ...)
}
reorder_within &lt;- function(x, by, within, fun = mean, sep = &quot;___&quot;, ...) {
  new_x &lt;- paste(x, within, sep = sep)
  stats::reorder(new_x, by, FUN = fun)
}</code></pre>
<div id="import-datatable-properties" class="section level2">
<h2>Import &amp; datatable properties</h2>
<pre class="r"><code>## load sparse matrix format all data
counts_PlateA &lt;- read_count_output(&quot;data/counts/PlateA/featurecounts&quot;, name = &quot;featurecounts&quot;)

meta_wellID &lt;- read_excel(&quot;data/config/meta_wellplate_corrected_switch_back.xlsx&quot;)
meta_Abs &lt;- read_excel(&quot;data/config/meta_antibody.xlsx&quot;)


  
## reshape counts in data table format
counts.dtbl &lt;- data.frame(counts_PlateA) %&gt;% 
  mutate(Barcodename = rownames(counts_PlateA)) %&gt;%
  select(Barcodename, everything()) %&gt;%
  gather(&quot;sample_BC&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateA)+1)) %&gt;%
  filter(counts &gt;1) %&gt;% ## Remove undetected proteins counts
   left_join(meta_Abs)%&gt;%
  unique() %&gt;%
  filter(!is.na(target_Full_Name)) %&gt;% #remove proteins not in metadata%&gt;%
  left_join(subset(meta_wellID, Plate == &quot;Plate_A&quot;)) %&gt;% ## Add sample metadata
  filter(!is.na(sample_ID))

## Calculate per-sample properties
data_properties &lt;- counts.dtbl %&gt;%
  unique()%&gt;%
  group_by(sample_BC) %&gt;%
  summarize(nCount = sum(counts), 
            nProt = n()) %&gt;%
  left_join(subset(meta_wellID, Plate == &quot;Plate_A&quot;)) %&gt;%
  filter(!is.na(buffer))## Add sample metadata </code></pre>
<p>The full panel of mapped barcodes and 96 wells are present in the
data table. After removing non-detected barcodes and proteins nmot
present in metadata file (not conjugated) we will explore the data
properties. First have a look at the total counts and number of detected
protein barcodes per sample:</p>
<pre class="r"><code>options(repr.plot.width=4, repr.plot.height=4)

ggplot(data_properties , aes(log10(nCount), nProt)) +
  geom_point(alpha = 1, size = 1) +
 # facet_wrap(~stimulus)+
  theme_bw() +
  labs(title = &quot;Correlation total counts and \nnumber of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;)</code></pre>
<p><strong>Figure 1.</strong> Correlation between Total counts and
number of detected protein barcodes. We observe many more detected
proteins than present in the staining. Note that we removed all counts
that were not in antibody metadata files. Sequencing depth of several
samples is very high (between 0.1M and 1M reads).</p>
<p>As we have a 96-wells plate, we can also have a look at total counts
per plate detected:</p>
<pre class="r"><code>options(repr.plot.width=8, repr.plot.height=3)

raw_map(data = log10(data_properties$nCount), well = data_properties$well_nr, 
        plate = 96, size = 6) +
  scale_fill_viridis(expression(&quot;Total counts Log&quot;[10])) +
  theme_bw()</code></pre>
<p><strong>Figure 2.</strong> Plate overview of total counts. We can see
several patterns of samples with high total counts. This is expected, as
in these wells we have the stained HBL1 cells. In other wells we do not
expect counts as these are different types of negative controls (MQ,
no-stain, no-cell).</p>
<p>To gain insight into effect of buffer conditions on number signal in
all different samples, we first have a look at not-normalized (raw)
counts per buffer per sample:</p>
<pre class="r"><code>ggplot(data_properties , aes(buffer, nCount, color = as.factor(replicate))) +
  geom_jitter(alpha = 0.8, size = 1, width = 0.1) +
  facet_wrap(~description_cell_treat, ncol = 4)+
  theme_bw() +
  RotatedAxis() +
 # scale_y_log10() +
  scale_color_colorblind(&quot;Replicate&quot;)</code></pre>
<p><strong>Figure 3.</strong> Per sample total counts. There are several
samples with high counts: as expected the stained HBL1 cells. The
negative samples (no-stain, MQ and no-cell have relatively low counts.)
Also the Jurkat T-cells have low counts.</p>
<p>To visualize better differences in samples with lower counts we can
plot log10 scale:</p>
<pre class="r"><code>ggplot(data_properties , aes(buffer, nCount, color = as.factor(replicate))) +
  geom_jitter(alpha = 0.8, size = 1, width = 0.1) +
  facet_wrap(~description_cell_treat, ncol = 4)+
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
               labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  labs(x = &quot;Buffer&quot;, y = expression(&quot;Total counts Log&quot;[10])) +
  scale_color_colorblind(&quot;Replicate&quot;)</code></pre>
<p><strong>Figure 3.log </strong> Per sample total counts on log-scale.
The negative control samples show around 1000-10.000 counts. The T-cell
line (Jurkat cells) show slightly higher counts(above 10.000 counts),
and the HBL1 cells show positive signal between 10.000 and 1M counts (2
orders of magnitude above background). There are buffer-depended effects
on background binding on the Jurkat cells (lowest with QuRIEseq, Highest
with CITE-seq, rest similar).</p>
</div>
<div id="signal-over-background" class="section level2">
<h2>Signal over background</h2>
<pre class="r"><code>data_properties.signalBackground &lt;- filter(data_properties, cell_line ==&quot;Jurkat&quot;) %&gt;%
  group_by(buffer ) %&gt;%
  summarise(background.jurkat.av = mean(nCount))
  
data_properties &lt;- left_join(data_properties, data_properties.signalBackground) %&gt;%
  mutate(signal_over_background = nCount/background.jurkat.av)</code></pre>
<pre class="r"><code>ggplot(data_properties , aes(buffer, signal_over_background, color = as.factor(replicate))) +
  geom_jitter(alpha = 0.8, size = 1, width = 0.1) +
  facet_wrap(~description_cell_treat, ncol = 4)+
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
               labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  # labs(x = &quot;Buffer&quot;, y = expression(&quot;Total counts Log&quot;[10])) +
  scale_color_colorblind(&quot;Replicate&quot;)</code></pre>
<p><strong>Figure 4.</strong></p>
<pre class="r"><code>data_properties.signalBackground.nocell &lt;- filter(data_properties, description ==&quot;noCell&quot;) %&gt;%
  group_by(buffer ) %&gt;%
  summarise(background.nocell.av = mean(nCount))
  
data_properties &lt;- left_join(data_properties, data_properties.signalBackground.nocell) %&gt;%
  mutate(signal_over_background.nocell = nCount/background.nocell.av)</code></pre>
<pre class="r"><code>ggplot(data_properties , aes(buffer, signal_over_background.nocell, color = as.factor(replicate))) +
  geom_jitter(alpha = 0.8, size = 1, width = 0.1) +
  facet_wrap(~description_cell_treat, ncol = 4)+
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
               labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  # labs(x = &quot;Buffer&quot;, y = expression(&quot;Total counts Log&quot;[10])) +
  scale_color_colorblind(&quot;Replicate&quot;)</code></pre>
<p><strong>Figure 4.</strong></p>
</div>
<div id="per-antibody-barcode-analysis" class="section level2">
<h2>Per-antibody barcode analysis</h2>
<pre class="r"><code>counts.dtbl.notpresentAbs &lt;- filter(counts.dtbl, target_easy_name_unique == &quot;NA_0&quot;)
counts.dtbl.Abs &lt;- filter(counts.dtbl, target_easy_name_unique != &quot;NA_0&quot;)</code></pre>
<pre class="r"><code>ggplot(counts.dtbl.Abs, aes(reorder(target_easy_name_unique ,counts ), counts , color = as.factor(description_cell_treat), shape = as.factor(replicate))) +
     # geom_boxplot(outlier.colour=&quot;black&quot;, outlier.shape=16,
     #         outlier.size=0.5, notch=FALSE) +
  geom_point(alpha = 0.7)+
     scale_x_reordered() +
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
 scale_y_log10(
   breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
   labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
 annotation_logticks(sides = &#39;l&#39;) +
  labs(x=&quot;protein&quot;)+
  theme_bw() +
  facet_wrap(~buffer, ncol = 2)+
  RotatedAxis()+
  scale_color_colorblind()</code></pre>
<p><strong>Figure5.</strong> Distribution raw counts per Ab-barcode.
Non-normalized counts show same patterns across all antibodies.</p>
<pre class="r"><code>ggplot(counts.dtbl.notpresentAbs, aes(reorder(Barcodename ,counts ), counts , color = as.factor(description_cell_treat), shape = as.factor(replicate))) +
     # geom_boxplot(outlier.colour=&quot;black&quot;, outlier.shape=16,
     #         outlier.size=0.5, notch=FALSE) +
  geom_point(alpha = 0.7)+
     scale_x_reordered() +
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
 # scale_y_log10(
 #   breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
 #   labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
 #  annotation_logticks(sides = &#39;l&#39;) +
  labs(x=&quot;protein&quot;)+
  theme_bw() +
  facet_wrap(~buffer, ncol = 2, scales = &quot;free_x&quot;)+
  RotatedAxis()+
  scale_color_colorblind()</code></pre>
<p><strong>Figure5.2</strong> Distribution raw counts per barcode not
present in experiment. Are these extra contaminants?</p>
<p>As there are multiple barcodes per antibody, we can add-up all these
values. Then we will have a look at HBL1 PBS and aIg samples, normalize
per staining the replicates using average scaling factor of non-phospho
antibodies CD19, ERK12, IkB and PARP.</p>
<pre class="r"><code>## First sum all counts from contaminant barcodes

counts.dtbl.Abs.sum &lt;- counts.dtbl.Abs %&gt;%
  ungroup() %&gt;%
  group_by(target_easy_name, description, description_cell_treat, buffer, cell_line, treatment, replicate) %&gt;%
  summarise(counts = sum(counts), .groups = &quot;keep&quot;) %&gt;%
  ungroup()

## Select stained cells
counts.dtbl.Abs.sum &lt;- counts.dtbl.Abs.sum %&gt;%
  filter(cell_line != &quot;MQ&quot; &amp; cell_line != &quot;noCell&quot; &amp; cell_line != &quot;Jurkat&quot;) %&gt;%
  filter(description_cell_treat != &quot;HBL1_aIg_nostain&quot; &amp; description_cell_treat != &quot;HBL1_PBS_nostain&quot;)  %&gt;%
  group_by(buffer, cell_line) %&gt;%
  mutate(counts_median = median(counts),
         counts_norm_median = counts/counts_median)%&gt;%
  ungroup()

# counts.nonphospho &lt;- counts.dtbl.Abs.sum %&gt;%
#   filter(target_easy_name == &quot;CD19&quot; |target_easy_name == &quot;ERK1_2&quot; |target_easy_name == &quot;IkB&quot; |target_easy_name == &quot;PARP&quot; )%&gt;%
#   ungroup() %&gt;%
#   group_by(buffer, description) %&gt;%
#   summarise(counts.total.nonphospho = sum(counts), .groups = &quot;keep&quot;) %&gt;%
#   ungroup() %&gt;%
#   group_by(buffer) %&gt;%
#   mutate(scaling.factor = counts.total.nonphospho/median(counts.total.nonphospho))


extra &lt;- counts.dtbl.Abs.sum %&gt;%
  filter(target_easy_name == &quot;CD19&quot; |target_easy_name == &quot;ERK1_2&quot; |target_easy_name == &quot;IkB&quot; |target_easy_name == &quot;PARP&quot; )%&gt;%
  ungroup() %&gt;%
  group_by(buffer, target_easy_name) %&gt;%
  mutate(scaling.factor = counts/median(counts))

counts.nonphospho &lt;- counts.dtbl.Abs.sum %&gt;%
  filter(target_easy_name == &quot;CD19&quot; |target_easy_name == &quot;ERK1_2&quot; |target_easy_name == &quot;IkB&quot; |target_easy_name == &quot;PARP&quot; )%&gt;%
  ungroup() %&gt;%
  group_by(buffer, target_easy_name) %&gt;%
  mutate(scaling.factor = counts/median(counts),
         scaling.factor.geo = counts/exp(mean(log((counts))))) %&gt;%
  ungroup()%&gt;%
  group_by(buffer, description)%&gt;%
  summarise(scaling.factor = mean(scaling.factor), .groups = &quot;keep&quot;,
            scaling.factor.geo = mean(scaling.factor.geo))

counts.dtbl.Abs.sum &lt;- counts.dtbl.Abs.sum %&gt;%
  left_join(counts.nonphospho) %&gt;%
  ungroup() %&gt;%
  mutate(counts_norm_household = counts/scaling.factor,
         counts_norm_household.geo = counts/scaling.factor.geo)%&gt;%
mutate(treatment = fct_relevel(treatment, 
            &quot;PBS&quot;, &quot;aIg&quot;),
       description_cell_treat = fct_relevel(description_cell_treat, &quot;HBL1_PBS&quot;, &quot;HBL1_aIg&quot;)) 

# counts.PARP &lt;- counts.dtbl.Abs.sum %&gt;%
#   filter(target_easy_name == &quot;PARP&quot; )%&gt;%
#   select(description, buffer,count__PARP =counts)%&gt;%
#   ungroup() %&gt;%
#   group_by(buffer) %&gt;%
#   mutate(scaling.factor.PARP = count__PARP/median(count__PARP))
# 
# 
# counts.dtbl.Abs.sum &lt;- counts.dtbl.Abs.sum %&gt;%
#   left_join(counts.PARP) %&gt;%
#   ungroup() %&gt;%
#   mutate(counts_norm_PARP = counts/scaling.factor.PARP)</code></pre>
<pre class="r"><code>ggplot(counts.dtbl.Abs.sum, aes(buffer, counts_norm_household , color = description_cell_treat, shape = as.factor(replicate), group = description_cell_treat)) +
  geom_point(alpha = 0.7, position = position_dodge(width = 0.5))+
  
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
 # scale_y_log10(
 #   breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
 #   labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
 # annotation_logticks(sides = &#39;l&#39;) +
  labs(x=&quot;&quot;)+
  theme_bw() +
  facet_wrap(~target_easy_name, ncol = 4)+
  RotatedAxis()+
  scale_color_colorblind()</code></pre>
<p><strong>Figure 6.</strong> Per-buffer normalized counts using Avg
scaling factor of 4 non-phospho proteins. Effect of aIg is hard to
see.</p>
<pre class="r"><code>ggplot(counts.dtbl.Abs.sum, aes(buffer, counts_norm_household.geo , color = description_cell_treat, shape = as.factor(replicate), group = description_cell_treat)) +
  geom_point(alpha = 0.7, position = position_dodge(width = 0.5))+
  
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
 # scale_y_log10(
 #   breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
 #   labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
 # annotation_logticks(sides = &#39;l&#39;) +
  labs(x=&quot;&quot;)+
  theme_bw() +
  facet_wrap(~target_easy_name, ncol = 4)+
  RotatedAxis()+
  scale_color_colorblind()</code></pre>
<p><strong>Figure 6.geo </strong> Per-buffer normalized counts using Avg
scaling factor of 4 non-phospho proteins. Effect of aIg is hard to
see.</p>
<p>To visualize effect of aIg, we calculate the fold-change and with
t-test determine p-value:</p>
<pre class="r"><code>FoldChanges &lt;- counts.dtbl.Abs.sum %&gt;%
  ungroup() %&gt;%
  group_by(treatment, buffer,target_easy_name) %&gt;%
  summarise(values = list(counts_norm_household), .groups = &quot;keep&quot;) %&gt;%
  ungroup()%&gt;%
  group_by(target_easy_name, buffer) %&gt;%
  summarise(fc = c(mean(values[[2]]) /mean(values[[1]])),
            lfc = log2(mean(values[[2]]) /mean(values[[1]])),
            p_value= t.test(values[[1]],values[[2]])$p.value, .groups = &quot;keep&quot;) %&gt;%
  ungroup()%&gt;%
  mutate(sign = ifelse(p_value &lt;0.05, &quot;sign&quot;, &quot;nonsign&quot;))

FoldChanges.geo &lt;- counts.dtbl.Abs.sum %&gt;%
  ungroup() %&gt;%
  group_by(treatment, buffer,target_easy_name) %&gt;%
  summarise(values = list(counts_norm_household.geo), .groups = &quot;keep&quot;) %&gt;%
  ungroup()%&gt;%
  group_by(target_easy_name, buffer) %&gt;%
  summarise(fc = c(mean(values[[2]]) /mean(values[[1]])),
            lfc = log2(mean(values[[2]]) /mean(values[[1]])),
            p_value= t.test(values[[1]],values[[2]])$p.value, .groups = &quot;keep&quot;) %&gt;%
  ungroup()%&gt;%
  mutate(sign = ifelse(p_value &lt;0.05, &quot;sign&quot;, &quot;nonsign&quot;))</code></pre>
<pre class="r"><code>ggplot(FoldChanges, aes(buffer, fc, color = sign)) +
  geom_point(alpha = 0.7)+
     scale_x_reordered() +
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
  labs(x=&quot;&quot;,y = &quot;Fold Change (aIg/PBS)&quot;)+
  theme_bw() +
  facet_wrap(~target_easy_name, ncol = 4)+
  RotatedAxis()+
  scale_color_colorblind(&quot;pval &lt; 0.05&quot;) +
  geom_hline(yintercept = 1)</code></pre>
<p><strong>Figure 7</strong> Fold-change aIg vs PBS. Not many
significant increased phospho-proteins. It seems QuRIE-seq buffer gives
best fold-change. This is the case even though overall raw counts are
much lower ass seen in figure 6. I’m not sure yet if this normalization
&amp; analysis is the best, and it is rather odd that the Ab-seq with
seemingly best signal over technical background has no significant
effect of aIg.</p>
<pre class="r"><code>ggplot(FoldChanges.geo, aes(buffer, fc, color = sign)) +
  geom_point(alpha = 0.7)+
     scale_x_reordered() +
     theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5,colour=&#39;black&#39;, size =6)) +
  labs(x=&quot;&quot;,y = &quot;Fold Change (aIg/PBS)&quot;)+
  theme_bw() +
  facet_wrap(~target_easy_name, ncol = 4)+
  RotatedAxis()+
  scale_color_colorblind(&quot;pval &lt; 0.05&quot;) +
  geom_hline(yintercept = 1)</code></pre>
<p><strong>Figure 7.geo</strong> Fold-change aIg vs PBS. Not many
significant increased phospho-proteins. It seems QuRIE-seq buffer gives
best fold-change. This is the case even though overall raw counts are
much lower ass seen in figure 6. I’m not sure yet if this normalization
&amp; analysis is the best, and it is rather odd that the Ab-seq with
seemingly best signal over technical background has no significant
effect of aIg.</p>
<pre class="r"><code># library(ggpubr)
# library(rstatix)
 
# stat.test &lt;- counts.dtbl.Abs.sum %&gt;%
#   group_by(buffer, target_easy_name) %&gt;%
#   t_test(counts_norm_household ~ treatment) %&gt;%
#   adjust_pvalue(method = &quot;bonferroni&quot;) %&gt;%
#   add_significance(&quot;p.adj&quot;)
# stat.test &lt;- stat.test %&gt;%
#   add_xy_position(fun = &quot;mean_sd&quot;, x = &quot;buffer&quot;, dodge = 0.8)

# 
# # Create bar plots of means
# ggbarplot(counts.dtbl.Abs.sum, x = &quot;buffer&quot;, y = &quot;counts_norm_household&quot;, 
#           facet.by =  &quot;target_easy_name&quot;,
#           add = c(&quot;mean_sd&quot;, &quot;jitter&quot;),
#           color = &quot;treatment&quot;, palette = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;),
#            position = position_dodge(0.8)) +
#   # stat_pvalue_manual(
#   # stat.test,  label = &quot;p&quot;, 
#   # tip.length = 0, hide.ns = TRUE
#   # )+
#   theme_bw() +
#   RotatedAxis()+
#   labs(x=&quot;&quot;)


# 
# df.summary &lt;- counts.dtbl.Abs.sum %&gt;%
#   group_by(buffer,treatment,target_easy_name) %&gt;%
#   summarise(
#     sd = sd(counts_norm_household, na.rm = TRUE),
#     counts_norm_household = mean(counts_norm_household)
#   )
# 
# 
# ggplot(df.summary, aes(buffer, counts_norm_household)) +
#   geom_jitter(data =(counts.dtbl.Abs.sum), aes(buffer, counts_norm_household), position = position_jitter(0.05), color = &quot;darkgray&quot;) + 
#   facet_wrap(~target_easy_name)+
#   geom_errorbar(
#     aes(ymin = counts_norm_household-sd, ymax = counts_norm_household+sd, color = treatment),
#     position = position_dodge(0.5), width = 0.2
#     )+
#   geom_point(aes(color = treatment), position = position_dodge(0.3)) +
#   scale_color_manual(values = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
