<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-08-06" />

<title>Chapter supplementary figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapter IDseq tech</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="y_chapter_main.html">Figs main</a>
</li>
<li>
  <a href="z_chapter_suppl.html">Figs suppl</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="e_IDseq_ann.html">Annotation</a>
    </li>
    <li>
      <a href="DS078_IDseq_analysis.html">DS078 analysis</a>
    </li>
    <li>
      <a href="DS091_IDseq_analysis.html">DS091 analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Chapter supplementary figures</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-08-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapter_IDseq_tech/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240806code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240806)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240806code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240806)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong31f8444">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 31f8444
</a>
</p>
</div>
<div id="strongRepositoryversionstrong31f8444"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 31f8444.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/DS075_combined_analysis.R
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/BB014_LibraryIndex/
    Untracked:  data/DS027_FirstIDseq/
    Untracked:  data/DS055_LibraryKlenowAmp/
    Untracked:  data/DS056_LibraryAmp1/
    Untracked:  data/DS059_LibraryByproducts/
    Untracked:  data/DS072_LibraryAmp2/
    Untracked:  data/DS075_BarcodeCheck/
    Untracked:  data/DS078_Blocking1/
    Untracked:  data/DS080_Blocking2/
    Untracked:  data/DS084_BioInhibition/
    Untracked:  data/DS091_ImmunostainingBio/
    Untracked:  data/DSxxx_Conjugations/
    Untracked:  data/annotated_gels_batch1-10_digital.pptx
    Untracked:  output/BioAnalyzer_plots/
    Untracked:  output/DS075_BarcodeCheck/
    Untracked:  output/DS078_Blocking1/
    Untracked:  output/DS091_ImmunostainingBio/
    Untracked:  output/figures/
    Untracked:  output/network_visual/

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/c_flow_vis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/z_chapter_suppl.Rmd</code>) and
HTML (<code>docs/z_chapter_suppl.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
31f8444
</td>
<td>
mwitmond
</td>
<td>
2025-02-10
</td>
<td>
Revised ch3 figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
a92dfa0
</td>
<td>
mwitmond
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
4e7f883
</td>
<td>
mwitmond
</td>
<td>
2024-08-06
</td>
<td>
Publish initial project files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)
PANEL_labels &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;,&quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

textsize_small &lt;- theme(text = element_text(size = 8, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 9)
)
textsize_medium &lt;- theme(text = element_text(size = 10, family = &quot;sans&quot;, colour = &quot;black&quot;),
                         strip.text.x = element_text(size = 10), 
                         plot.title = element_text(size = 10)
)</code></pre>
<pre class="r"><code>colors_nord &lt;- c(&quot;#2e3440&quot;, &quot;#3b4252&quot;, &quot;#434c5e&quot;, &quot;#4c566a&quot;, 
                 &quot;#d8dee9&quot;, &quot;#e5e9f0&quot;, &quot;#eceff4&quot;, 
                 &quot;#8fbcbb&quot;, &quot;#88c0d0&quot;, &quot;#81a1c1&quot;, &quot;#5e81ac&quot;, 
                 &quot;#bf616a&quot;, &quot;#d08770&quot;, &quot;#ebcb8b&quot;, &quot;#a3be8c&quot;, &quot;#b48ead&quot;)

colors_toll_bright &lt;- c(&quot;#4477AA&quot;, &quot;#66CCEE&quot;, &quot;#228833&quot;, &quot;#CCBB44&quot;, &quot;#EE6677&quot;, &quot;#AA3377&quot;, &quot;#BBBBBB&quot;)
colors_toll_mute &lt;- c(&quot;#332288&quot;, &quot;#88CCEE&quot;, &quot;#44AA99&quot;, &quot;#117733&quot;, &quot;#999933&quot;, &quot;#DDCC77&quot;, &quot;#CC6677&quot;, &quot;#882255&quot;, &quot;#AA4499&quot;, &quot;#DDDDDD&quot;)
colors_toll_br &lt;- c(&quot;#364B9A&quot;, &quot;#4A7BB7&quot;, &quot;#6EA6CD&quot;, &quot;#98CAE1&quot;, &quot;#C2E4EF&quot;, &quot;#EAECCC&quot;, &quot;#FEDA8B&quot;, &quot;#FDB366&quot;, &quot;#F67A4B&quot;, &quot;#DD3D2D&quot;, &quot;#A50026&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)
colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

# Variable-specific colors
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#737373&quot;, &quot;anti-Ig&quot; = &quot;#2cb142&quot;, &quot;H2O2&quot; = &quot;#66CCEE&quot;, 
                 &quot;Basal&quot; = &quot;#737373&quot;, &quot;Activated (anti-Ig)&quot; = &quot;#2cb142&quot;, &quot;Activated (H2O2)&quot; = &quot;#66CCEE&quot;)
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#df65b0&quot;, &quot;DAUDI&quot; = &quot;#41b6c4&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;)
# colors_cell_alt &lt;- c(&quot;HBL1&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;)
colors_block_buf &lt;- c(&quot;ID-seq&quot; = &quot;#88CCEE&quot;, 
                      &quot;RAID&quot; = &quot;#44AA99&quot;, 
                      &quot;QuRIE-seq&quot; = &quot;#117733&quot;, 
                      &quot;Ab-seq&quot; = &quot;#AA4499&quot;, 
                      &quot;CITE-seq&quot; = &quot;#CC6677&quot;, 
                      &quot;NEAT-seq&quot; = &quot;#A50026&quot;)
colors_stain_conc &lt;- c(&quot;0.5 ug/mL&quot; = &quot;#3f007d&quot;, 
                          &quot;0.25 ug/mL&quot; = &quot;#54278f&quot;, 
                          &quot;0.125 ug/mL&quot; = &quot;#807dba&quot;, 
                          &quot;0.063 ug/mL&quot; = &quot;#9e9ac8&quot;, 
                          &quot;0.031 ug/mL&quot; = &quot;#bcbddc&quot;, 
                          &quot;0 ug/mL&quot; = &quot;#d9d9d9&quot;)
colors_stain_cell &lt;- c(&quot;0M (no cell)&quot; = &quot;#d9d9d9&quot;, &quot;0.25M&quot; = &quot;#9e9ac8&quot;, &quot;0.5M&quot; = &quot;#54278f&quot;)
colors_stain_volume &lt;- c(&quot;20 uL&quot; = &quot;#fdbe85&quot;, &quot;30 uL&quot; = &quot;#fd8d3c&quot;, &quot;40 uL&quot; = &quot;#e6550d&quot;)</code></pre>
</div>
<div id="suppl.-figure-s1" class="section level2">
<h2>Suppl. Figure S1</h2>
<p><em>Figure S1: Detailed B-cell signaling network.</em></p>
<p>Input: - BioRender schematic figures</p>
<div id="panels" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
figS1_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS1_panelA

# PNG
png_figS1_panelA &lt;- image_read(&quot;output/figures/non_R_figs/ch3_Bcell_network_full_202411.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_figS1_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS1_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
png_figS1_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS1_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  figS1_panelA,
  filename = &quot;output/figures/suppl_figS1.pdf&quot;,
  width = 6,
  height = 6,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS1_panelA,
  filename = &quot;output/figures/suppl_figS1_place.jpg&quot;,
  width = 6,
  height = 6,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S1: Detailed B-cell signaling network.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS1&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 10594146 565.8   17081781 912.3 12781065 682.6
Vcells 18367274 140.2   71773003 547.6 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s2" class="section level2">
<h2>Suppl. Figure S2</h2>
<p><em>Figure S2: Antibody-DNA conjugates after 2 years of storage at
-20C.</em></p>
<p>Input: - SDS-PAGE gel images</p>
<div id="panels-1" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-1" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
figS2_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS2_panelA

# PNG
png_figS2_panelA &lt;- image_read(&quot;output/figures/non_R_figs/conjugation_immediately.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_figS2_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS2_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
figS2_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS2_panelB

# PNG
png_figS2_panelB &lt;- image_read(&quot;output/figures/non_R_figs/conjugation_after2years.png&quot;) %&gt;%
  image_ggplot()
png_figS2_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS2_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-1" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS2 &lt;- plot_grid(png_figS2_panelA, png_figS2_panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10)
figS2</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS2_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS2,
  filename = &quot;output/figures/suppl_figS2.pdf&quot;,
  width = 6,
  height = 5.7,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS2,
  filename = &quot;output/figures/suppl_figS2.jpg&quot;,
  width = 6,
  height = 5.7,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S2: Antibody-DNA conjugates after 2 years of storage at
-20C.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS2&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 10632394 567.9   17081781 912.3 14901174 795.9
Vcells 23709166 180.9   71773003 547.6 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s3" class="section level2">
<h2>Suppl. Figure S3</h2>
<p><em>Figure S3: Examples of clean and contaminated Ab
barcodes.</em></p>
<p>Input:</p>
<ul>
<li>DS075 ID-seq data</li>
</ul>
<div id="data" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS075 (counts per barcode per well)
figS3_data_id &lt;- read_csv(&quot;output/DS075_BarcodeCheck/IDseq_ann/IDseq_data_sample.csv&quot;)</code></pre>
<pre class="r"><code>figS3_BC_color &lt;- c(&quot;correct&quot; = &quot;#2cb142&quot;, &quot;false&quot; = &quot;black&quot;) #&quot;#2cb142&quot; &quot;#74c476&quot;</code></pre>
</div>
<div id="panels-2" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-2" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
set.seed(8448)
figS3A_data_id &lt;- figS3_data_id %&gt;%
  dplyr::filter(counts_percent &gt; 0.1) %&gt;%
  group_by(expected_BC_long) %&gt;%
  mutate(total_prot = n(),
         expected_BC_label = paste(&quot;Expected: BC&quot;, expected_BC), 
         expected_BC_color = case_when(measured_BC_long == expected_BC_long ~ &quot;correct&quot;, 
                                       .default = &quot;false&quot;)) %&gt;%
  ungroup() %&gt;%
  dplyr::filter(total_prot == 1 &amp; expected_BC &lt; 100) %&gt;%
  # sample_n(6)
  subset(expected_BC %in% c(52, 59, 72, 75, 90, 97))

# Figure
figS3_panelA &lt;- ggplot(figS3A_data_id) +
  geom_point(aes(factor(measured_BC), counts_percent, color = expected_BC_color)) + 
  facet_wrap(vars(factor(expected_BC_label)), scales = &quot;free_x&quot;, nrow = 3) + 
  coord_cartesian(ylim = c(0, 100)) +
  scale_color_manual(values = figS3_BC_color) +
  labs(x = &quot;Measured Ab barcode&quot;, y = &quot;Percentage of total counts&quot;) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
figS3_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS3_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-1" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
set.seed(8448)
figS3B_data_id &lt;- figS3_data_id %&gt;%
  dplyr::filter(counts_percent &gt; 2.5) %&gt;%
  group_by(expected_BC_long) %&gt;%
  mutate(total_prot = n(),
         expected_BC_label = paste(&quot;Expected: BC&quot;, expected_BC), 
         expected_BC_color = case_when(measured_BC_long == expected_BC_long ~ &quot;correct&quot;, 
                                       .default = &quot;false&quot;)) %&gt;%
  ungroup() %&gt;%
  dplyr::filter(total_prot &gt; 1 &amp; expected_BC &lt; 100) %&gt;%
  # sample_n(6)
  subset(expected_BC %in% c(3, 7, 14, 22, 53, 56, 74, 83, 96))

# Figure
figS3_panelB &lt;- ggplot(figS3B_data_id) +
  geom_point(aes(factor(measured_BC), counts_percent, color = expected_BC_color)) + 
  facet_wrap(vars(factor(expected_BC_label)), scales = &quot;free_x&quot;, nrow = 3) +
  # facet_wrap(~expected_BC_long, scales = &quot;free_x&quot;) + 
  coord_cartesian(ylim = c(0, 100)) +
  scale_color_manual(values = figS3_BC_color) +
  labs(x = &quot;Measured Ab barcode&quot;, y = &quot;Percentage of total counts&quot;) +
  theme_bw() +
  # theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  theme(panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
figS3_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS3_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-2" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS3 &lt;- plot_grid(figS3_panelA, figS3_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1.5), label_size = 10)
figS3</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS3_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS3,
  filename = &quot;output/figures/suppl_figS3.pdf&quot;,
  width = 6,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS3,
  filename = &quot;output/figures/suppl_figS3.jpg&quot;,
  width = 6,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S3: Examples of clean and contaminated Ab
barcodes.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS3&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 10743496 573.8   17081781 912.3 17081781 912.3
Vcells 18650657 142.3   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s4" class="section level2">
<h2>Suppl. Figure S4</h2>
<p><em>Figure S4: Immunostaining conditions in DAUDI treated with
anti-Ig.</em></p>
<p>Input:</p>
<ul>
<li>DS091 ID-seq data</li>
</ul>
<div id="data-1" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS091 (counts per well/sample and mean per condition)
figS4_data_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/IDseq_data_sample.csv&quot;)
# figS4_mean_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/IDseq_data_condition.csv&quot;)</code></pre>
<pre class="r"><code>figS4_stain_conc &lt;- c(&quot;0.5 ug/mL&quot;, &quot;0.25 ug/mL&quot;, &quot;0.125 ug/mL&quot;, &quot;0.063 ug/mL&quot;, &quot;0.031 ug/mL&quot;, &quot;0 ug/mL&quot;)</code></pre>
</div>
<div id="panels-3" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-3" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis of the effect of staining conc on
H2O2 vs PBS</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1 cells: 2 stimuli, 6 staining conc, 3 staining cell nr, 3
staining volumes</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: staining_conc_text (ref: 0.5 ug/mL) + stimulus (
ref: PBS) + staining_conc_text:stimulus</p></li>
</ul>
<pre class="r"><code># Select data
figS4A_data &lt;- figS4_data_id %&gt;%
  dplyr::filter(cell_line == &quot;DAUDI&quot; &amp; staining_cells == &quot;0.25M&quot; &amp; staining_volume != &quot;20 uL&quot; &amp; counts &gt;= 1)

# Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- figS4A_data  %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(figS4A_data[, c(1:16)]) %&gt;% 
  distinct() %&gt;%
  mutate(staining_conc_text = factor(staining_conc_text, levels = figS4_stain_conc))

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ staining_conc_text + stimulus + staining_conc_text:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$staining_conc_text &lt;- relevel(dds$staining_conc_text, ref = &quot;0.5 ug/mL&quot;)
dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_aIg &lt;- results(dds, name = &quot;stimulus_anti.Ig_vs_PBS&quot;, alpha = 0.05)
# summary(results_H2O2, padj = 0.05)
results_conc_0.25 &lt;- results(dds, name = &quot;staining_conc_text_0.25.ug.mL_vs_0.5.ug.mL&quot;, alpha = 0.05)
results_conc_0.125 &lt;- results(dds, name = &quot;staining_conc_text_0.125.ug.mL_vs_0.5.ug.mL&quot;, alpha = 0.05)
results_conc_0.063 &lt;- results(dds, name = &quot;staining_conc_text_0.063.ug.mL_vs_0.5.ug.mL&quot;, alpha = 0.05)
results_conc_0.031 &lt;- results(dds, name = &quot;staining_conc_text_0.031.ug.mL_vs_0.5.ug.mL&quot;, alpha = 0.05)
results_conc_0 &lt;- results(dds, name = &quot;staining_conc_text_0.ug.mL_vs_0.5.ug.mL&quot;, alpha = 0.05)

results_stim_0.25 &lt;- results(dds, name = &quot;staining_conc_text0.25.ug.mL.stimulusanti.Ig&quot;, alpha = 0.05)
results_stim_0.125 &lt;- results(dds, name = &quot;staining_conc_text0.125.ug.mL.stimulusanti.Ig&quot;, alpha = 0.05)
results_stim_0.063 &lt;- results(dds, name = &quot;staining_conc_text0.063.ug.mL.stimulusanti.Ig&quot;, alpha = 0.05)
results_stim_0.031 &lt;- results(dds, name = &quot;staining_conc_text0.031.ug.mL.stimulusanti.Ig&quot;, alpha = 0.05)
results_stim_0 &lt;- results(dds, name = &quot;staining_conc_text0.ug.mL.stimulusanti.Ig&quot;, alpha = 0.05)

list_results &lt;- list(results_aIg, 
                     results_conc_0.25, results_conc_0.125, results_conc_0.063, results_conc_0.031, results_conc_0, 
                     results_stim_0.25, results_stim_0.125, results_stim_0.063, results_stim_0.031, results_stim_0)

# Data transformation
# Regularized log transform
rld_DAUDI &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.2)
names_volc &lt;- c(&quot;volc_aIg&quot;, 
                &quot;volc_conc_0.25&quot;, &quot;volc_conc_0.125&quot;, &quot;volc_conc_0.063&quot;, &quot;volc_conc_0.031&quot;, &quot;volc_conc_0&quot;, 
                &quot;volc_stim_0.25&quot;, &quot;volc_stim_0.125&quot;, &quot;volc_stim_0.063&quot;, &quot;volc_stim_0.031&quot;, &quot;volc_stim_0&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_aIg, 
                          volc_conc_0.25, volc_conc_0.125, volc_conc_0.063, volc_conc_0.031, volc_conc_0, 
                          volc_stim_0.25, volc_stim_0.125, volc_stim_0.063, volc_stim_0.031, volc_stim_0)
names_heat &lt;- c(&quot;heat_aIg&quot;, 
                &quot;heat_conc_0.25&quot;, &quot;heat_conc_0.125&quot;, &quot;heat_conc_0.063&quot;, &quot;heat_conc_0.031&quot;, &quot;heat_conc_0&quot;, 
                &quot;heat_stim_0.25&quot;, &quot;heat_stim_0.125&quot;, &quot;heat_stim_0.063&quot;, &quot;heat_stim_0.031&quot;, &quot;heat_stim_0&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
<pre class="r"><code># Data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_combi_DAUDI &lt;- list(heat_aIg#, 
                        # heat_conc_0.25, heat_conc_0.125, heat_conc_0.063, heat_conc_0.031, heat_conc_0
                        # heat_stim_0.25, heat_stim_0.125, heat_stim_0.063, heat_stim_0.031, heat_stim_0
                        ) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  left_join(figS4_data_id[, 17:24]) %&gt;%
  dplyr::filter(modification == &quot;phospho&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
figS4A_info_DAUDI &lt;- figS4A_data %&gt;%
  dplyr::select(plate_well, staining_volume, staining_conc_text, stim_clean) %&gt;%
  distinct() %&gt;%
  # dplyr::filter(cell_line == &quot;DAUDI&quot; &amp;inhibitor == &quot;iBTK&quot; &amp; inhib_conc %in% fig2_conc_all &amp; !(stim_clean == &quot;Basal&quot; &amp; inhib_conc %in% fig2_conc_all[2:5])) %&gt;%
  mutate(stim_clean = factor(stim_clean, levels = c(&quot;Basal&quot;, &quot;Activated (H2O2)&quot;)), 
         staining_conc_text = factor(staining_conc_text, levels = figS4_stain_conc), 
         staining_conc_label = factor(str_replace(staining_conc_text, &quot; &quot;, &quot;\n&quot;), levels = c(&quot;0.5\nug/mL&quot;, &quot;0.25\nug/mL&quot;, &quot;0.125\nug/mL&quot;, &quot;0.063\nug/mL&quot;, &quot;0.031\nug/mL&quot;, &quot;0\nug/mL&quot;))) %&gt;%
  as.data.frame()
rownames(figS4A_info_DAUDI) &lt;- figS4A_info_DAUDI$plate_well
figS4A_info_DAUDI &lt;- dplyr::select(figS4A_info_DAUDI, -c(plate_well))

# Get the rld transformed data for the heatmap and scale per row
figS4A_data_heat_DAUDI &lt;- t(scale(t(assay(rld_DAUDI)[c(heat_combi_DAUDI$target_nospace), rownames(figS4A_info_DAUDI)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
figS4A_ann_colors &lt;- list(#Cells = colors_stain_cell,
                         &quot;Staining volume&quot; = colors_stain_volume, 
                         &quot;Panel conc.&quot; = colors_stain_conc, 
                         Stimulus = colors_stim)
figS4A_heat_legend &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 7), labels_gp = gpar(fontsize = 7), 
                          grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

figS4A_col_ann &lt;- HeatmapAnnotation(#Cells = figS4A_info_DAUDI$staining_cells, 
                                   &quot;Staining volume&quot; = figS4A_info_DAUDI$staining_volume, 
                                   &quot;Panel conc.&quot; = figS4A_info_DAUDI$staining_conc_text,
                                   Stimulus = figS4A_info_DAUDI$stim_clean, 
                                   simple_anno_size = unit(2.5, &quot;mm&quot;), 
                                   annotation_name_gp = gpar(fontsize = 7),
                                   col = figS4A_ann_colors, 
                                   annotation_legend_param = list(#Cells = list(title = &quot;Cells&quot;, 
                                                                                   # title_gp = gpar(fontsize = 6), 
                                                                                   # labels_gp = gpar(fontsize = 6), 
                                                                                   # grid_width = unit(2, &quot;mm&quot;), 
                                                                                   # grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Staining volume&quot; = list(title = &quot;Staining\nvolume&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2.5, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Panel conc.&quot; = list(title = &quot;Panel\nconcentration&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2.5, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  Stimulus = list(title = &quot;Stimulus&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2.5, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;))), 
                                   show_legend = c(TRUE, TRUE, TRUE))

figS4_panelA &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  figS4A_data_heat_DAUDI,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  # treeheight_row = 20,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  top_annotation = figS4A_col_ann,
  annotation_colors = figS4A_ann_colors,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = figS4A_info_DAUDI$staining_conc_label,
  # row_split = heat_combi_DAUDI$modification,
  row_gap = unit(1, &quot;mm&quot;), 
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 6,
  row_title_gp = gpar(fontsize = 6),
  column_title_gp = gpar(fontsize = 6), 
  annotation_names_col = gpar(fontsize = 7),
  border_color = NA,
  heatmap_legend_param = figS4A_heat_legend,
  # annotation_legend = FALSE,
)) +
  # ggtitle(&quot;          Signaling activation in DAUDI under different staining conditions&quot;) +
  textsize_medium
figS4_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS4_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-3" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS4_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS4_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS4_panelA,
  filename = &quot;output/figures/suppl_figS4.pdf&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS4_panelA,
  filename = &quot;output/figures/suppl_figS4.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S4: Immunostaining conditions in DAUDI treated with
anti-Ig.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS4&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11036214 589.4   17081781 912.3 17081781 912.3
Vcells 19326300 147.5   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s5" class="section level2">
<h2>Suppl. Figure S5</h2>
<p><em>Figure S5: Comparison of phospho flow cytometry and ID-seq
measurements.</em></p>
<p>Input:</p>
<ul>
<li><p>DS091 phosphoflow data</p></li>
<li><p>DS091 ID-seq data</p></li>
</ul>
<div id="data-2" class="section level3">
<h3>Data</h3>
<pre class="r"><code># Phosphoflow data DS091
figS5_data_pf &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/flow_ann/flow_data_DS091.csv&quot;)

# ID-seq data DS091 (counts per well/sample and mean per condition)
figS5_data_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/bio_IDseq_data_sample.csv&quot;)
figS5_mean_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/bio_IDseq_data_condition.csv&quot;)</code></pre>
<pre class="r"><code>figS5_stim &lt;- c(&quot;PBS&quot;, &quot;anti-Ig&quot;, &quot;H2O2&quot;)
figS5_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated (anti-Ig)&quot;, &quot;Activated (H2O2)&quot;)
figS5_cell &lt;- c(&quot;DAUDI&quot;, &quot;HBL1&quot;)
figS5_cell_stim &lt;- c(&quot;DAUDI - Basal&quot;, &quot;DAUDI - Activated (anti-Ig)&quot;, &quot;HBL1 - Basal&quot;, &quot;HBL1 - Activated (H2O2)&quot;)
figS5_cell_stim_label &lt;- c(&quot;DAUDI\nBasal&quot;, &quot;DAUDI\nActivated\n(anti-Ig)&quot;, &quot;HBL1\nBasal&quot;, &quot;HBL1\nActivated\n(H2O2)&quot;)

figS5_proteins_pf &lt;- c(&quot;pCD79a (Y182)&quot;, &quot;pSYK (Y525/Y526)&quot;) #&quot;pPLCy2 (Y759)&quot;
figS5_proteins_id &lt;- c(&quot;CD79a_Y182&quot;, &quot;SYK_Y525_Y526&quot;) #PLCy2_Y759&quot;
figS5_proteins_id_label &lt;- c(&quot;CD79a_Y182&quot; = &quot;pCD79a (Y182)&quot;, &quot;SYK_Y525_Y526&quot; = &quot;pSYK (Y525/Y526)&quot;) #PLCy2_Y759&quot;</code></pre>
</div>
<div id="panels-4" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-4" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
figS5A_data &lt;- figS5_data_pf %&gt;%
  # subset(protein == &quot;pCD79a (Y182)&quot;)
  subset(protein %in% figS5_proteins_pf)

# Figure
figS5_panelA &lt;- ggplot(figS5A_data, aes(x = fluorescence)) +
  geom_density_ridges(
    aes(y = factor(description, levels = rev(figS5_cell_stim), labels = rev(figS5_cell_stim_label)), fill = factor(stim_clean, levels = figS5_stim_clean)),
    scale = 1, 
    alpha = 0.5
  ) +
  facet_wrap(vars(factor(protein, levels = figS5_proteins_pf)), nrow = 1) +
  scale_x_logicle() + # logicle scale instead of log10 scale
  scale_fill_manual(values = colors_stim, name = &quot;Treatment&quot;) +
  labs(x = &quot;Fluorescent intensity&quot;, y = &quot;&quot;) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;right&quot;) +
  textsize_small
figS5_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS5_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-2" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
figS5B_data &lt;- figS5_data_id %&gt;%
  # subset(target_nospace == &quot;CD79a_Y182&quot;)
  subset(target_nospace %in% figS5_proteins_id)
figS5B_mean &lt;- figS5_mean_id %&gt;%
  # subset(target_nospace == &quot;CD79a_Y182&quot;)
  subset(target_nospace %in% figS5_proteins_id)

# Figure
figS5_panelB &lt;- ggplot() +
  geom_jitter(data = figS5B_data,
              aes(x = factor(description_sample, levels = figS5_cell_stim, labels = figS5_cell_stim_label),
                  y = counts_norm,
                  color = factor(stim_clean, levels = figS5_stim_clean)),
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = figS5B_mean,
                aes(x = factor(description_sample, levels = figS5_cell_stim, labels = figS5_cell_stim_label), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stim_clean, levels = figS5_stim_clean),
                    color = factor(stim_clean, levels = figS5_stim_clean)),
                width = 0.35, size = 0.75) +
  facet_wrap(vars(factor(target_nospace, levels = figS5_proteins_id, labels = figS5_proteins_id_label)), nrow = 1, scales = &quot;free_y&quot;) +
  scale_color_manual(values = colors_stim, name = &quot;Treatment&quot;) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;right&quot;) +
  textsize_small
figS5_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS5_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-4" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS5 &lt;- plot_grid(figS5_panelA, figS5_panelB, labels = PANEL_labels[c(1:2)], ncol = 1, rel_heights = c(1.2, 1), label_size = 10, align = &quot;v&quot;)
figS5</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS5_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS5,
  filename = &quot;output/figures/suppl_figS5.pdf&quot;,
  width = 4,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS5,
  filename = &quot;output/figures/suppl_figS5.jpg&quot;,
  width = 4,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S5: Comparison of phospho flow cytometry and ID-seq
measurements.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS5&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11073864 591.5   17081781 912.3 17081781 912.3
Vcells 22488438 171.6   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s6" class="section level2">
<h2>Suppl. Figure S6</h2>
<p><em>Figure S6: Volcano visualisation of activated signaling compared
to basal signaling.</em></p>
<p>Input:</p>
<ul>
<li>DS091 ID-seq data</li>
</ul>
<div id="data-3" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS091 (counts per well/sample)
figS6_data_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/bio_IDseq_data_sample.csv&quot;)</code></pre>
<pre class="r"><code>figS6_cells &lt;- c(&quot;DAUDI&quot;, &quot;HBL1&quot;)
figS6_stim &lt;- c(&quot;Basal&quot;, &quot;Activated (anti-Ig)&quot;, &quot;Activated (H2O2)&quot;)
figS6_stim_label &lt;- c(&quot;Basal&quot;, &quot;Activated\n(anti-Ig)&quot;, &quot;Activated\n(H2O2)&quot;)
figS6_cell_stim &lt;- c(&quot;DAUDI - Basal&quot;, &quot;DAUDI - Activated (anti-Ig)&quot;, &quot;HBL1 - Basal&quot;, &quot;HBL1 - Activated (H2O2)&quot;)
figS6_cell_stim_label &lt;- c(&quot;DAUDI basal&quot;, &quot;DAUDI activated\n(anti-Ig)&quot;, &quot;HBL1 basal&quot;, &quot;HBL1 activated\n(H2O2)&quot;)

figS6_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;replicate&quot;, &quot;description_sample&quot;, &quot;description_sample_rep&quot;, &quot;description_staining&quot;)</code></pre>
</div>
<div id="panels-5" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-5" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis of activated (anti-Ig) vs basal
signaling in DAUDI</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>DAUDI cells: 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: stim_clean (ref: Basal)</p></li>
</ul>
<pre class="r"><code># Select data
figS6A_data &lt;- figS6_data_id %&gt;%
  dplyr::filter(cell_line == &quot;DAUDI&quot; &amp; counts &gt;= 1)

# Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- figS6A_data  %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(figS6A_data[, c(1:16)]) %&gt;% 
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$staining_conc_text &lt;- relevel(dds$staining_conc_text, ref = &quot;0.5 ug/mL&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_DAUDI &lt;- results(dds, name = &quot;stim_clean_Activated..anti.Ig._vs_Basal&quot;, alpha = 0.05)
# summary(results_DAUDI, padj = 0.05)
list_results &lt;- list(results_DAUDI)

# Data transformation
# Regularized log transform
rld_DAUDI &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.1)
names_volc &lt;- c(&quot;volc_DAUDI&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}</code></pre>
<pre class="r"><code># Select data
figS6A_data_volc &lt;- volc_DAUDI 
figS6A_data_volc$wrapper &lt;- &quot;Burkitt&#39;s model (DAUDI)\nanti-Ig vs PBS&quot;

fig6A_stat &lt;- as.data.frame(figS6A_data_volc) %&gt;%
  dplyr::filter(diff_express == &quot;UP&quot;) %&gt;%
  dplyr::summarise(log2FC = mean(log2FoldChange)) %&gt;%
  pull(log2FC)
print(&quot;DAUDI mean log2FC&quot;)</code></pre>
<pre><code>[1] &quot;DAUDI mean log2FC&quot;</code></pre>
<pre class="r"><code>fig6A_stat</code></pre>
<pre><code>[1] 0.4827916</code></pre>
<pre class="r"><code>print(&quot;2^(mean log2FC)&quot;)</code></pre>
<pre><code>[1] &quot;2^(mean log2FC)&quot;</code></pre>
<pre class="r"><code>(2^fig6A_stat)</code></pre>
<pre><code>[1] 1.397445</code></pre>
<pre class="r"><code># Figure
figS6_panelA &lt;- ggplot(data = figS6A_data_volc,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 2,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(figS6A_data_volc$log2FoldChange &lt; 0, -0.5 - figS6A_data_volc$log2FoldChange, 0.6 - figS6A_data_volc$log2FoldChange)
  ) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  facet_wrap(vars(factor(wrapper))) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  # coord_cartesian(xlim = c(-0.5, 1.5), ylim = c(0, 325)) +
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, legend.justification = &quot;left&quot;, panel.grid.minor = element_blank(), panel.grid.major = element_blank(), strip.text.x = element_text(size = 8)) + 
  guides(color = guide_legend(ncol = 1, byrow = TRUE))
figS6_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS6_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-3" class="section level4">
<h4>Panel B</h4>
<p>Differential expression analysis of activated (anti-Ig) vs basal
signaling in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1 cells: 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: stim_clean (ref: Basal)</p></li>
</ul>
<pre class="r"><code># Select data
figS6B_data &lt;- figS6_data_id %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; counts &gt;= 1)

# Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- figS6B_data  %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(figS6B_data[, c(1:16)]) %&gt;% 
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1 &lt;- results(dds, name = &quot;stim_clean_Activated..H2O2._vs_Basal&quot;, alpha = 0.05)
# summary(results_HBL1, padj = 0.05)
list_results &lt;- list(results_HBL1)

# Data transformation
# Regularized log transform
rld_HBL1 &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.25)
names_volc &lt;- c(&quot;volc_HBL1&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.2)))
}</code></pre>
<pre class="r"><code># Select data
figS6B_data_volc &lt;- volc_HBL1
figS6B_data_volc$wrapper &lt;- &quot;DLBCL model (HBL1)\nH2O2 vs PBS&quot;

fig6B_stat &lt;- as.data.frame(figS6B_data_volc) %&gt;%
  dplyr::filter(diff_express == &quot;UP&quot;) %&gt;%
  dplyr::summarise(log2FC = mean(log2FoldChange)) %&gt;%
  pull(log2FC)
print(&quot;HBL1 mean log2FC&quot;)</code></pre>
<pre><code>[1] &quot;HBL1 mean log2FC&quot;</code></pre>
<pre class="r"><code>fig6B_stat</code></pre>
<pre><code>[1] 0.888017</code></pre>
<pre class="r"><code>print(&quot;2^(mean log2FC)&quot;)</code></pre>
<pre><code>[1] &quot;2^(mean log2FC)&quot;</code></pre>
<pre class="r"><code>(2^fig6B_stat)</code></pre>
<pre><code>[1] 1.850631</code></pre>
<pre class="r"><code># Figure
figS6_panelB &lt;- ggplot(data = figS6B_data_volc,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  max.overlaps = 11,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(figS6B_data_volc$log2FoldChange &lt; 0, -1.05 - figS6B_data_volc$log2FoldChange, 1.55 - figS6B_data_volc$log2FoldChange)
  ) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.2), log2(1.2)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  facet_wrap(vars(factor(wrapper))) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  # coord_cartesian(xlim = c(-0.5, 1.5), ylim = c(0, 325)) +
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, legend.justification = &quot;left&quot;, panel.grid.minor = element_blank(), panel.grid.major = element_blank(), strip.text.x = element_text(size = 8)) + 
  guides(color = guide_legend(ncol = 1, byrow = TRUE))
figS6_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS6_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-5" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS6 &lt;- plot_grid(figS6_panelA, figS6_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1), label_size = 10, align = &quot;h&quot;)
figS6</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS6_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS6,
  filename = &quot;output/figures/suppl_figS6.pdf&quot;,
  width = 6,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS6,
  filename = &quot;output/figures/suppl_figS6.jpg&quot;,
  width = 6,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S6: Volcano visualisation of activated signaling compared
to basal signaling.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS6&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11101634 592.9   17081781 912.3 17081781 912.3
Vcells 19439632 148.4   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s7" class="section level2">
<h2>Suppl. Figure S7</h2>
<p><em>Figure S7: Additional example proteins for signaling activation
in DAUDI and HBL1.</em></p>
<p>Input:</p>
<ul>
<li>DS091 ID-seq data</li>
</ul>
<div id="data-4" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS091 (counts per well/sample and mean per condition)
figS7_data_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/bio_IDseq_data_sample.csv&quot;)
figS7_mean_id &lt;- read_csv(&quot;output/DS091_ImmunostainingBio/IDseq_ann/bio_IDseq_data_condition.csv&quot;)</code></pre>
<pre class="r"><code>figS7_cells &lt;- c(&quot;DAUDI&quot;, &quot;HBL1&quot;)
figS7_stim &lt;- c(&quot;Basal&quot;, &quot;Activated (anti-Ig)&quot;, &quot;Activated (H2O2)&quot;)
figS7_stim_label &lt;- c(&quot;Basal&quot;, &quot;Activated\n(anti-Ig)&quot;, &quot;Activated\n(H2O2)&quot;)
figS7_cell_stim &lt;- c(&quot;DAUDI - Basal&quot;, &quot;DAUDI - Activated (anti-Ig)&quot;, &quot;HBL1 - Basal&quot;, &quot;HBL1 - Activated (H2O2)&quot;)
figS7_cell_stim_label &lt;- c(&quot;DAUDI basal&quot;, &quot;DAUDI activated\n(anti-Ig)&quot;, &quot;HBL1 basal&quot;, &quot;HBL1 activated\n(H2O2)&quot;)</code></pre>
</div>
<div id="panels-6" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-6" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
# figS7A_prot &lt;- c(&quot;BTK_Y551&quot;, &quot;CREB_S133&quot;, &quot;PYK2_Y402&quot;, &quot;S6_S240_S244&quot;, &quot;VAV1_Y174&quot;)
figS7A_prot &lt;- c(&quot;BLNK_Y84&quot;, &quot;BTK_Y223&quot;, &quot;LYN_Y397&quot;, &quot;SHP2_Y580&quot;, &quot;SYK_Y525_Y526&quot;)
figS7A_prot_label &lt;- c(&quot;BLNK\n(Y84)&quot;, &quot;BTK\n(Y223)&quot;, &quot;LYN\n(Y397)&quot;, &quot;SHP2\n(Y580)&quot;, &quot;SYK\n(Y525/Y526)&quot;)
figS7A_data &lt;- figS7_data_id %&gt;%
  dplyr::filter(cell_line == &quot;DAUDI&quot;) %&gt;%
  subset(target_nospace %in% figS7A_prot)
figS7A_mean &lt;- figS7_mean_id %&gt;%
  dplyr::filter(cell_line == &quot;DAUDI&quot;) %&gt;%
  subset(target_nospace %in% figS7A_prot)

# Figure
figS7_panelA &lt;- ggplot() +
  geom_jitter(data = figS7A_data,
              aes(x = factor(target_nospace, levels = figS7A_prot, labels = figS7A_prot_label),
                  y = counts_norm,
                  color = factor(stim_clean, levels = figS7_stim)),
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = figS7A_mean,
                aes(x = factor(target_nospace, levels = figS7A_prot, labels = figS7A_prot_label),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stim_clean, levels = figS7_stim),
                    color = factor(stim_clean, levels = figS7_stim)),
                width = 0.35, size = 0.75) +
  scale_color_manual(values = colors_stim, breaks = figS7_stim, labels = figS7_stim_label, name = &quot;Treatment&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;) +
  theme_bw() +
  # RotatedAxis() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.position = &quot;bottom&quot;, legend.justification = &quot;left&quot;, axis.text.x = element_text(angle = 45, hjust = 0.5, vjust = 0.75)) +
  textsize_small
figS7_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS7_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-4" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
figS7B_prot &lt;- c(&quot;BTK_Y551&quot;, &quot;CREB_S133&quot;, &quot;PYK2_Y402&quot;, &quot;S6_S240_S244&quot;, &quot;VAV1_Y174&quot;)
figS7B_prot_label &lt;- c(&quot;BTK\n(Y551)&quot;, &quot;CREB\n(S133)&quot;, &quot;PYK2\n(Y402)&quot;, &quot;S6\n(S240/S244)&quot;, &quot;VAV1\n(Y174)&quot;)
# figS7B_prot &lt;- c(&quot;BLNK_Y84&quot;, &quot;BTK_Y223&quot;, &quot;LYN_Y397&quot;, &quot;SHP2_Y580&quot;, &quot;SYK_Y525_Y526&quot;)
figS7B_data &lt;- figS7_data_id %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot;) %&gt;%
  subset(target_nospace %in% figS7B_prot)
figS7B_mean &lt;- figS7_mean_id %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot;) %&gt;%
  subset(target_nospace %in% figS7B_prot)

# Figure
figS7_panelB &lt;- ggplot() +
  geom_jitter(data = figS7B_data,
              aes(x = factor(target_nospace, levels  =figS7B_prot, labels = figS7B_prot_label),
                  y = counts_norm,
                  color = factor(stim_clean, levels = figS7_stim)),
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = figS7B_mean,
                aes(x = factor(target_nospace, levels  =figS7B_prot, labels = figS7B_prot_label),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stim_clean, levels = figS7_stim),
                    color = factor(stim_clean, levels = figS7_stim)),
                width = 0.35, size = 0.75) +
  scale_color_manual(values = colors_stim, breaks = figS7_stim, labels = figS7_stim_label, name = &quot;Treatment&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;) +
  theme_bw() +
  RotatedAxis() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.position = &quot;bottom&quot;, legend.justification = &quot;left&quot;, axis.text.x = element_text(angle = 45, hjust = 0.5, vjust = 0.75)) +
  # guides(color = guide_legend(ncol = 1, byrow = TRUE)) +
  textsize_small
figS7_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS7_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-6" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS7 &lt;- plot_grid(figS7_panelA, figS7_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1), label_size = 10, align = &quot;h&quot;)
figS7</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS7_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS7,
  filename = &quot;output/figures/suppl_figS7.pdf&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS7,
  filename = &quot;output/figures/suppl_figS7.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S7: Additional example proteins for signaling activation
in DAUDI and HBL1.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS7&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11099872 592.8   17081781 912.3 17081781 912.3
Vcells 19436359 148.3   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s8" class="section level2">
<h2>Suppl. Figure S8</h2>
<p><em>Figure S8: Quality control of ID-seq data.</em></p>
<p>Input:</p>
<ul>
<li>DS091 ID-seq data</li>
</ul>
<div id="data-5" class="section level3">
<h3>Data</h3>
<pre class="r"><code># Load sparse matrix format all data
figS8_counts_PlateA &lt;- read_count_output(&quot;data/DS091_ImmunostainingBio/counts/PlateA/featurecounts&quot;, name = &quot;featurecounts&quot;)
figS8_meta_wellID &lt;- read_csv(&quot;data/DS091_ImmunostainingBio/config/DynSign.091_wellBC_metadata_seq.csv&quot;)
figS8_meta_Abs &lt;- read_excel(&quot;data/DS091_ImmunostainingBio/config/BulkIDseq_Ab_info_seq_analysis_202212.xlsx&quot;)
figS8_comp_matrix &lt;- read_csv(&quot;data/DS091_ImmunostainingBio/BC_compensation_matrix.csv&quot;)

# Reshape counts in data table format
figS8_counts_dtbl &lt;- data.frame(figS8_counts_PlateA) %&gt;% 
  mutate(barcode_name = rownames(figS8_counts_PlateA)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(figS8_counts_PlateA)+1)) %&gt;%
  dplyr::filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(figS8_meta_Abs) %&gt;%
  unique() %&gt;%
  left_join(subset(figS8_meta_wellID, plate == &quot;A&quot;)) %&gt;% # Add sample metadata
  dplyr::filter(!is.na(experiment)) %&gt;%
  arrange(plate, well) %&gt;%
  mutate(plate_well = paste0(plate, &quot;_&quot;, well))</code></pre>
<pre class="r"><code>figS8_conc &lt;- c(0.5, 0.25, 0.125, 0.063, 0.031, 0)
figS8_conc_text &lt;- c(&quot;0.5 ug/mL&quot;, &quot;0.25 ug/mL&quot;, &quot;0.125 ug/mL&quot;, &quot;0.063 ug/mL&quot;, &quot;0.031 ug/mL&quot;, &quot;0 ug/mL&quot;)
figS8_conc_labels &lt;- c(&quot;0.5\nug/mL&quot;, &quot;0.25\nug/mL&quot;, &quot;0.125\nug/mL&quot;, &quot;0.063\nug/mL&quot;, &quot;0.031\nug/mL&quot;, &quot;0\nug/mL&quot;)</code></pre>
</div>
<div id="panels-7" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-7" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
# Calculate per-sample properties
figS8A_data &lt;- figS8_counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(counts), 
            nProt = n()) %&gt;%
  left_join(figS8_meta_wellID)

# Figure
figS8_panelA &lt;- ggplot(figS8A_data, aes(log10(nCount), nProt, color = staining_conc_text)) +
  geom_point(alpha = 0.9, size = 1) +
  facet_wrap(vars(factor(experiment, labels = &quot;Before analysis\n(raw counts)&quot;))) +
  coord_cartesian(xlim = c(2.7, 6)) +
  scale_color_manual(values = colors_stain_conc, name = &quot;Staining conc.&quot;) +
  labs(x = expression(&quot;Total counts (log&quot;[10]*&quot;)&quot;),
       y = &quot;Detected Ab barcodes&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(0.35, &quot;cm&quot;), legend.spacing.y = unit(0.01, &quot;cm&quot;), legend.box.spacing = unit(0.05, &quot;cm&quot;), panel.grid.minor = element_blank(), strip.text.x = element_text(size = 7.5), legend.title = element_text(size = 7))
figS8_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS8_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-5" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
figS8_data_comp &lt;- figS8_counts_dtbl %&gt;%
  dplyr::select(plate, well, barcode_name, counts) %&gt;%
  left_join(figS8_comp_matrix, by = join_by(barcode_name == BC_to_correct)) #, multiple = &quot;all&quot;

# Get counts of OG BC
figS8_data_comp_OG &lt;- figS8_data_comp %&gt;%
  dplyr::filter(barcode_name == OG_BC) %&gt;%
  dplyr::select(plate, well, OG_BC, counts) %&gt;%
  dplyr::rename(counts_OG_BC = counts)

# Calculate false reads
figS8_data_false &lt;- left_join(figS8_data_comp, figS8_data_comp_OG) %&gt;% 
  mutate(false_counts = (counts_OG_BC / percent_OG_BC) * percent_BC_to_correct) %&gt;%
  dplyr::filter(barcode_name != OG_BC) %&gt;%
  group_by(plate, well, barcode_name, counts) %&gt;%
  summarize(total_false_counts = sum(false_counts, na.rm = TRUE))

figS8B_counts_dtbl &lt;- figS8_counts_dtbl %&gt;%
  left_join(figS8_data_false) %&gt;% 
  mutate(total_false_counts = replace_na(total_false_counts, 0), 
         true_counts = as.integer(counts - total_false_counts), 
         percent_true = (true_counts / counts) * 100) %&gt;%
  dplyr::filter(!is.na(target_nospace)) # Remove barcodes not in Ab metadata (not conjugated/added in panel)

# Calculate per-sample properties
figS8B_data &lt;- figS8B_counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(true_counts), 
            nProt = n()) %&gt;%
  left_join(figS8_meta_wellID)

# Figure
figS8_panelB &lt;- ggplot(figS8B_data, aes(log10(nCount), nProt, color = staining_conc_text)) +
  geom_point(alpha = 0.9, size = 1) +
  facet_wrap(vars(factor(experiment, labels = &quot;After correction&quot;))) +
  coord_cartesian(xlim = c(2.7, 6)) +
  scale_color_manual(values = colors_stain_conc, name = &quot;Staining conc.&quot;) +
  labs(x = expression(&quot;Total counts (log&quot;[10]*&quot;)&quot;),
       y = &quot;Detected Ab barcodes&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(0.35, &quot;cm&quot;), legend.spacing.y = unit(0.01, &quot;cm&quot;), legend.box.spacing = unit(0.05, &quot;cm&quot;), panel.grid.minor = element_blank(), strip.text.x = element_text(size = 7.5, margin = margin(0.3, 0, 0.3, 0, &quot;cm&quot;)), legend.title = element_text(size = 7))
figS8_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS8_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-c" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
figS8_scaling_factors &lt;- figS8B_counts_dtbl %&gt;%
  dplyr::filter(cell_line %in% c(&quot;HBL1&quot;, &quot;DAUDI&quot;)) %&gt;%
  ungroup() %&gt;%
  group_by(cell_line, target_nospace) %&gt;%
  mutate(true_scaling_factor_geo = true_counts / exp(mean(log((true_counts))))) %&gt;%
  ungroup() %&gt;%
  group_by(cell_line, description_sample, description_staining) %&gt;%
  summarize(true_scaling_factor_geo = median(true_scaling_factor_geo, na.rm = TRUE))

figS8C_counts_dtbl_norm &lt;- figS8B_counts_dtbl %&gt;%
  dplyr::filter(cell_line %in% c(&quot;HBL1&quot;, &quot;DAUDI&quot;)) %&gt;%
  left_join(figS8_scaling_factors) %&gt;%
  ungroup() %&gt;%
  mutate(true_counts_norm_geo = true_counts / true_scaling_factor_geo) %&gt;%
  dplyr::filter(!is.na(target_nospace))

# Calculate per-sample properties
figS8C_data &lt;- figS8C_counts_dtbl_norm %&gt;%
  unique()%&gt;%
  group_by(well_BC_seq) %&gt;%
  summarize(nCount_norm = sum(true_counts_norm_geo), 
            nProt = n()) %&gt;%
  left_join(subset(figS8_meta_wellID, plate == &quot;A&quot;))

# Figure
figS8_panelC &lt;- ggplot(figS8C_data, aes(log10(nCount_norm), nProt, color = factor(staining_conc, levels = figS8_conc, labels = figS8_conc_text))) +
  geom_point(alpha = 0.9, size = 1) +
  facet_wrap(vars(factor(experiment, labels = &quot;After normalisation&quot;))) +
  coord_cartesian(xlim = c(2.7, 6)) +
  scale_color_manual(values = colors_stain_conc, labels = figS8_conc, name = &quot;Staining conc.\n(ug/mL)&quot;) +
  labs(x = expression(&quot;Total counts (log&quot;[10]*&quot;)&quot;),
       y = &quot;Detected Ab barcodes&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(0.35, &quot;cm&quot;), legend.spacing.y = unit(0.01, &quot;cm&quot;), legend.box.spacing = unit(0.05, &quot;cm&quot;), panel.grid.minor = element_blank(), strip.text.x = element_text(size = 7.5, margin = margin(0.3, 0, 0.3, 0, &quot;cm&quot;)), legend.title = element_text(size = 7), legend.text = element_text(size = 7.5))
figS8_panelC</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS8_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-d" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Select data
rng &lt;- range(log10(figS8A_data$nCount))

# Figure
figS8_panelD &lt;- raw_map(data = log10(figS8A_data$nCount), well = figS8A_data$well, plate = 96, size = 7) +
  scale_fill_viridis(expression(&quot;Total counts (log&quot;[10]*&quot;)&quot;), limits = c(rng[1], rng[2])) +
  theme_bw() +
  textsize_small
figS8_panelD</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS8_panelD-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-7" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS8_ABC &lt;- plot_grid(figS8_panelA, figS8_panelB, figS8_panelC, labels = PANEL_labels[c(1, 2, 3)], nrow = 1, rel_widths = c(1, 1, 1.4), label_size = 10)
figS8 &lt;- plot_grid(figS8_ABC, figS8_panelD, labels = c(NA, PANEL_labels[c(4)]), ncol = 1, rel_heights = c(1, 1.4), label_size = 10)
figS8</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS8_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS8,
  filename = &quot;output/figures/suppl_figS8.pdf&quot;,
  width = 6,
  height = 4.8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS8,
  filename = &quot;output/figures/suppl_figS8.jpg&quot;,
  width = 6,
  height = 4.8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S8: Quality control of ID-seq data.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS8&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11137907 594.9   17081781 912.3 17081781 912.3
Vcells 19521847 149.0   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s9" class="section level2">
<h2>Suppl. Figure S9</h2>
<p><em>Figure S9: Phospho flow cytometry gating strategy.</em></p>
<p>Input:</p>
<ul>
<li>DS091 phosphoflow data</li>
</ul>
<div id="panels-8" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-8" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
figS9_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS9_panelA

# PNG
png_figS9_panelA &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_debris_DS091.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_figS9_panelA</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS9_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-6" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
figS9_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS9_panelB

# PNG
png_figS9_panelB &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_singlets_DS091.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_figS9_panelB</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS9_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-c-1" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Placeholder
figS9_panelC &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS9_panelC

# PNG
png_figS9_panelC &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_live_DS091.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_figS9_panelC</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS9_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-8" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS9 &lt;- plot_grid(png_figS9_panelA, png_figS9_panelB, png_figS9_panelC, labels = PANEL_labels[c(1, 2, 3)], ncol = 3, rel_widths = c(1, 1, 1), label_size = 10)
figS9</code></pre>
<p><img src="figure/z_chapter_suppl.Rmd/figS9_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  figS9,
  filename = &quot;output/figures/suppl_figS9.pdf&quot;,
  width = 6,
  height = 2,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  figS9,
  filename = &quot;output/figures/suppl_figS9.jpg&quot;,
  width = 6,
  height = 2,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S9: Phospho flow cytometry gating strategy.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS9&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells 11160719 596.1   17081781 912.3 17081781 912.3
Vcells 21985786 167.8   57418403 438.1 89385421 682.0</code></pre>
</div>
</div>
<div id="suppl.-figure-s0" class="section level2">
<h2>Suppl. Figure S0</h2>
<p><em>Figure title.</em></p>
<p>Input: - BioRender schematic figures</p>
<ul>
<li><p>… phosphoflow data</p></li>
<li><p>… ID-seq data</p></li>
</ul>
<div id="data-6" class="section level3">
<h3>Data</h3>
</div>
<div id="panels-9" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-9" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
figS0_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# figS0_panelA

# PNG
png_figS0 &lt;- image_read(&quot;output/figures/non_R_figs/xxx.png&quot;) %&gt;%
  image_ggplot()
# png_figS0</code></pre>
</div>
<div id="panel-b-7" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
<div id="panel-c-2" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
</div>
<div id="combined-fig-9" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
figS0 &lt;- plot_grid(panelA, panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10)
figS0

# Save figure as pdf, jpg, and png
# ggsave(
#   figS0,
#   filename = &quot;output/figures/suppl_figS0.pdf&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  figS0,
  filename = &quot;output/figures/suppl_figS0.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   figS0,
#   filename = &quot;output/figures/suppl_figS0.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
<p><em>Figure title.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS0&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] png_0.1-8                   magick_2.8.3               
 [3] ggridges_0.5.6              forecast_8.23.0            
 [5] zoo_1.8-12                  ggcyto_1.32.0              
 [7] flowWorkspace_4.16.0        ncdfFlow_2.50.0            
 [9] BH_1.84.0-0                 flowCore_2.16.0            
[11] ashr_2.2-63                 apeglm_1.26.1              
[13] MASS_7.3-60.2               ggbreak_0.1.2              
[15] geomtextpath_0.1.4          ggh4x_0.2.8                
[17] cowplot_1.1.3               patchwork_1.2.0            
[19] greekLetters_1.0.2          ggpubr_0.6.0               
[21] ggrepel_0.9.5               scales_1.3.0               
[23] ggthemes_5.1.0              RColorBrewer_1.1-3         
[25] scico_1.5.0                 viridis_0.6.5              
[27] viridisLite_0.4.2           mmtable2_0.1.3             
[29] gt_0.10.1                   ggvenn_0.1.10              
[31] rgl_1.3.1                   plotly_4.10.4              
[33] ggcorrplot_0.1.4.1          corrplot_0.92              
[35] umap_0.2.10.0               ggplotify_0.1.2            
[37] dendsort_0.3.4              vsn_3.72.0                 
[39] ComplexHeatmap_2.20.0       platetools_0.1.7           
[41] kableExtra_1.4.0            DESeq2_1.44.0              
[43] SummarizedExperiment_1.34.0 Biobase_2.64.0             
[45] MatrixGenerics_1.16.0       matrixStats_1.3.0          
[47] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        
[49] IRanges_2.38.0              S4Vectors_0.42.0           
[51] BiocGenerics_0.50.0         Seurat_5.1.0               
[53] SeuratObject_5.0.2          sp_2.1-4                   
[55] rstatix_0.7.2               factoextra_1.0.7           
[57] FactoMineR_2.11             readxl_1.4.3               
[59] Matrix_1.7-0                lubridate_1.9.3            
[61] forcats_1.0.0               stringr_1.5.1              
[63] dplyr_1.1.4                 purrr_1.0.2                
[65] readr_2.1.5                 tidyr_1.3.1                
[67] tibble_3.2.1                ggplot2_3.5.1              
[69] tidyverse_2.0.0             workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             nnet_7.3-19             goftest_1.2-3          
  [4] DT_0.33                 TH.data_1.1-2           vctrs_0.6.5            
  [7] spatstat.random_3.2-3   digest_0.6.36           shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       SQUAREM_2021.1          reshape2_1.4.4         
 [16] httpuv_1.6.15           foreach_1.5.2           withr_3.0.0            
 [19] xfun_0.45               ggfun_0.1.5             survival_3.6-4         
 [22] memoise_2.0.1           hexbin_1.28.3           RProtoBufLib_2.16.0    
 [25] emmeans_1.10.2          systemfonts_1.1.0       ragg_1.3.2             
 [28] GlobalOptions_0.1.2     quantmod_0.4.26         pbapply_1.7-2          
 [31] promises_1.3.0          scatterplot3d_0.3-44    httr_1.4.7             
 [34] globals_0.16.3          fitdistrplus_1.1-11     ps_1.7.6               
 [37] rstudioapi_0.16.0       UCSC.utils_1.0.0        miniUI_0.1.1.1         
 [40] generics_0.1.3          base64enc_0.1-3         processx_3.8.4         
 [43] curl_5.2.1              zlibbioc_1.50.0         polyclip_1.10-6        
 [46] quadprog_1.5-8          GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [49] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [52] S4Arrays_1.4.1          preprocessCore_1.66.0   fracdiff_1.5-3         
 [55] hms_1.1.3               irlba_2.3.5.1           colorspace_2.1-0       
 [58] urca_1.3-4              ROCR_1.0-11             reticulate_1.38.0      
 [61] spatstat.data_3.1-2     Rgraphviz_2.48.0        magrittr_2.0.3         
 [64] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [67] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [70] XML_3.99-0.17           scattermore_1.2         xts_0.14.0             
 [73] RcppAnnoy_0.0.22        pillar_1.9.0            nlme_3.1-164           
 [76] iterators_1.0.14        compiler_4.4.1          RSpectra_0.16-1        
 [79] stringi_1.8.4           tensor_1.5              plyr_1.8.9             
 [82] crayon_1.5.3            abind_1.4-5             truncnorm_1.0-9        
 [85] gridGraphics_0.5-1      emdbook_1.3.13          locfit_1.5-9.10        
 [88] bit_4.0.5               sandwich_3.1-0          whisker_0.4.1          
 [91] codetools_0.2-20        multcomp_1.4-25         textshaping_0.4.0      
 [94] TTR_0.24.4              openssl_2.2.0           bslib_0.7.0            
 [97] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
[100] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
[103] tseries_0.10-56         cellranger_1.1.0        Rttf2pt1_1.3.12        
[106] leaps_3.2               knitr_1.47              utf8_1.2.4             
[109] clue_0.3-65             fs_1.6.4                listenv_0.9.1          
[112] ggsignif_0.6.4          estimability_1.5.1      callr_3.7.6            
[115] statmod_1.5.0           tzdb_0.4.0              svglite_2.1.3          
[118] pkgconfig_2.0.3         tools_4.4.1             cachem_1.1.0           
[121] numDeriv_2016.8-1.1     fastmap_1.2.0           rmarkdown_2.27         
[124] ica_1.0-3               broom_1.0.6             sass_0.4.9             
[127] coda_0.19-4.1           BiocManager_1.30.23     dotCall64_1.1-1        
[130] graph_1.82.0            carData_3.0-5           RANN_2.6.1             
[133] farver_2.1.2            yaml_2.3.8              cli_3.6.3              
[136] leiden_0.4.3.1          lifecycle_1.0.4         askpass_1.2.0          
[139] uwot_0.2.2              mvtnorm_1.2-5           backports_1.5.0        
[142] cytolib_2.16.0          BiocParallel_1.38.0     timechange_0.3.0       
[145] gtable_0.3.5            rjson_0.2.21            progressr_0.14.0       
[148] parallel_4.4.1          limma_3.60.3            jsonlite_1.8.8         
[151] RcppHNSW_0.6.0          bit64_4.0.5             multcompView_0.1-10    
[154] assertthat_0.2.1        Rtsne_0.17              yulab.utils_0.1.4      
[157] spatstat.utils_3.0-5    bdsmatrix_1.3-7         highr_0.11             
[160] jquerylib_0.1.4         timeDate_4032.109       lazyeval_0.2.2         
[163] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[166] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[169] XVector_0.44.0          rprojroot_2.0.4         gridExtra_2.3          
[172] flashClust_1.01-2       invgamma_1.1            igraph_2.0.3           
[175] extrafontdb_1.0         R6_2.5.1                labeling_0.4.3         
[178] bbmle_1.0.25.1          cluster_2.1.6           aplot_0.2.3            
[181] DelayedArray_0.30.1     tidyselect_1.2.1        xml2_1.3.6             
[184] car_3.1-2               future_1.33.2           munsell_0.5.1          
[187] KernSmooth_2.23-24      affyio_1.74.0           data.table_1.15.4      
[190] htmlwidgets_1.6.4       rlang_1.1.4             spatstat.sparse_3.1-0  
[193] extrafont_0.19          spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
